distribution/lib/Standard/Table/0.0.0-dev/src/Data/Column_Format.enso

```diff
@@ -0,0 +1,101 @@
+from Standard.Base import all
+import Standard.Base.Errors.Illegal_Argument.Illegal_Argument
+import Standard.Base.Errors.Illegal_State.Illegal_State
+
+import project.Data.Column.Column
+import project.Data.Data_Formatter.Data_Formatter
+import project.Data.Type.Storage
+
+from project.Data.Type.Value_Type import Value_Type
+from project.Internal.Java_Exports import make_string_builder
+
+polyglot java import java.lang.IllegalArgumentException
+polyglot java import java.time.temporal.UnsupportedTemporalTypeException
+polyglot java import org.enso.table.data.column.operation.map.MapOperationProblemBuilder
+polyglot java import org.enso.table.data.column.storage.Storage as Java_Storage
+polyglot java import org.enso.table.data.table.Column as Java_Column
+polyglot java import org.enso.table.operations.OrderBuilder
+
+## PRIVATE
+   Create a formatter for the given format string.
+   The `value` parameter has to have a `format` method that takes a format and
+   locale.
+make_value_formatter : Locale -> (Any -> Text)
+make_value_formatter locale = value-> format->
+    handle_illegal_argument_exception format <|
+        if format.is_nothing || format.is_empty then value.to_text else
+            value.format format locale
+## PRIVATE
+   Create a `Boolean` formatter that takes the format string as the second
+   parameter.
+make_boolean_formatter : (Boolean -> Text -> Text)
+make_boolean_formatter = bool-> format->
+    if format.is_nothing || format.is_empty then bool.to_text else
+        data_formatter = Data_Formatter.Value.with_format Value_Type.Boolean format
+        data_formatter.format bool
+
+## PRIVATE
+   Rethrow a Java IllegalArgumentException as an Illegal_Argument.
+handle_illegal_argument_exception : Text -> Any -> Any
+handle_illegal_argument_exception format_string ~action =
+    handler cause =
+        msg = cause.payload.getMessage + ' in \"' + format_string + '\"'
+        Error.throw (Illegal_Argument.Error msg)
+    Panic.catch IllegalArgumentException handler=handler <|
+        Panic.catch UnsupportedTemporalTypeException handler=handler action
+
+## PRIVATE
+   Iterate over a range, exiting early if the body produces an `Error`.
+each_propagate : Range -> (Number -> Any) -> Nothing ! Error
+each_propagate range function =
+    if range.step == 0 then Error.throw (Illegal_State.Error "A range with step = 0 is ill-formed.") else
+        end_condition = if range.step > 0 then (>=) else (<=)
+        go current =
+            if end_condition current range.end then Nothing else
+                result = function current
+                result.if_not_error <|
+                    @Tail_Call go current+range.step
+        go range.start
+
+## PRIVATE
+   Map a text-returning function over the column values, using Storage directly.
+   The output column has the same name as the input.
+map_over_storage : Column -> (Any -> Text) -> (Integer -> Any) -> Boolean -> Column
```

Comment:
these feel like they belong more general location but this is just a nit

Reply:
I agree; the moment I can use it a second time I'll move it somewhere.